src/components/EnvironmentOverride/DeploymentTemplateOverrideForm.tsx

```diff
@@ -228,7 +228,7 @@ export default function DeploymentTemplateOverrideForm({
             const deploymentTemplateResp = isConfigProtectionEnabled
                 ? await checkForProtectedLockedChanges()
                 : await api(+appId, +envId, payload)
-            if (deploymentTemplateResp.result.isLockConfigError && !saveEligibleChanges) {
+            if (deploymentTemplateResp.result?.isLockConfigError && !saveEligibleChanges) {
```

Comment:
The result field is optional and keeping in line with the code inside the block the optional chaining ought to be there.